FirebaseSessions/Sources/SessionGenerator.swift

```diff
@@ -39,25 +41,37 @@ class SessionGenerator {
   private var thisSession: SessionInfo?
   private var settings: SettingsProtocol
 
+  private var firstSessionId: String?
+  private var sessionIndex: Int32
+
   init(settings: SettingsProtocol) {
     self.settings = settings
+    // This will be incremented to 0 on the first generation
+    sessionIndex = -1
   }
 
   // Generates a new Session ID. If there was already a generated Session ID
   // from the last session during the app's lifecycle, it will also set the last Session ID
   func generateNewSession() -> SessionInfo {
     let newSessionId = UUID().uuidString.replacingOccurrences(of: "-", with: "").lowercased()
 
+    // If firstSessionId is set, use it. Otherwise set it to the
+    // first generated Session ID
+    let firstSessionId = self.firstSessionId ?? newSessionId
+    self.firstSessionId = firstSessionId
```

Comment:
Can we consolidate it to `self.firstSessionId = self.firstSessionId ?? newSessionId`. Is that an anti-pattern?

Reply:
Yeah I was trying to find a non-clunky way to write this. Since `self.firstSessionId` is a `String?`, I needed a way to define it as a `String` so I used a variable.

If I use `self.firstSessionId = self.firstSessionId ?? newSessionId`, then I need another `??` in the constructor for `SessionInfo`. I still think there's a better way to write this...

<img width="852" alt="Screenshot 2023-02-01 at 1 55 02 PM" src="https://user-images.githubusercontent.com/555046/216136808-1adb3010-6c0a-4d7e-9459-f04feb56a6a2.png">

Reply:
Maybe a ternary `self.firstSessionId = self.firstSessionId.isEmpty ? newSessionId : self.firstSessionId`, but I kindof hate ternarys